language/changes/6-phantom-type-params.md

```diff
@@ -0,0 +1,103 @@
+compile time. Previously, Move's type system didn't make a difference on whether a type parameter
+was phantom, but with this new feature, a struct's type parameter can be explicitly declared as
+phantom. A parameter declared as phantom is not considered when computing the
+[conditional abilities with generic types](3-abilities.md#Conditional-Abilities-with-Generic-Types).
+For this relaxed rule to be sound, Move's type system guarantees that a parameter declared as
+phantom is either not used at all in the struct definition, or it is only used as an argument to
+type parameters also declared as phantom.
+
+## Motivations
+
+A phantom type parameter may come in handy in a couple of situations. For example, the struct
+`Event::EventHandle<T>` in stdlib doesn't contain any field of type `T`, but uses the type parameter
+to ensure at the type level that the handle can only be used for messages of type `T`. Another
+notorious example is `Diem::Diem<CoinType>`, which is generic on a `CoinType` specifying the
+currency of the coin and allowing code to be written generically on any currency. As a last example,
+the capabilities in `Vault` also rely heavily on phantom type parameters.
+
+Previously, these examples required spurious ability annotations to satisfy the requirements of the
+abilities declared for each of the structs. For example, `Diem::Diem<CoinType>` is declared with the
+ability `key`, which required all arguments to `CoinType` to be declared with `store`. Specifically,
+the types `XUS::XUS` and `XDX::XDX` were forced to have a `store` annotation even though they were
+never stored in the database! More over the extra `store` requirement polluted the entire Diem
+Framework and many functions had to specify an extra `store` constraints on their type parameters.
+With the new feature, `Diem::Diem<CoinType>` now declares `CoinType` to be phantom which avoids the
+unnecessary `store` annotations.
+
+## Description
+
+### Declaration
+
+In a struct definition a type parameter can be declared as phantom by adding the `phantom` keyword
+before its declaration. If a type parameter is declared as phantom we say it is a phantom type
+parameter. When defining a struct, Move's type checker ensures that every phantom type parameter is
+either not used inside the struct definition or it is only used as an argument to a phantom type
+parameter.
+
+More formally, if a type is used as an argument to a parameter declared as phantom we say the type
+appears in phantom position. With this definition in place, the rule for the correct use of phantom
+parameters can be specified as follows: **A phantom type parameter can only appear in phantom
+position**.
+
+The following two examples show valid uses of phantom parameters. In the first one, the parameter
+`T1` is not used at all inside the struct definition. In the second one, the parameter `T1` is only
+used as an argument to a parameter also declared as phantom.
```

Comment:
I find this "declared as phantom" terminology awkward. I don't have a good suggestion for a direct replacement, but we could try to avoid using it more than necessary. For example, this sentence could say "...used as an argument to another phantom type parameter." Perhaps you could rewrite some of the other text with this terminology in a similar fashion?